diesel/src/query_builder/select_statement/mod.rs

```diff
@@ -176,26 +165,16 @@ impl<S, D, W, O, L, Of, G, DB> QueryFragment<DB>
         Ok(())
     }
 
-    fn collect_binds(&self, out: &mut DB::BindCollector) -> QueryResult<()> {
-        try!(self.distinct.collect_binds(out));
-        try!(self.select.collect_binds(&(), out));
-        try!(self.where_clause.collect_binds(out));
-        try!(self.group_by.collect_binds(out));
-        try!(self.order.collect_binds(out));
-        try!(self.limit.collect_binds(out));
-        try!(self.offset.collect_binds(out));
+    fn walk_ast(&self, pass: &mut AstPass<DB>) -> QueryResult<()> {
```

Comment:
What happens here for the is_safe_to_cache_prepared pass ?
If distinct is false, and then select is true ?

Ultimately, only the last one means something right ?
Maybe you should use `**result &= true/false` everywhere ?